### PR TITLE
[Improvement] Add information for featured exercises

### PR DESCRIPTION
**Pull request by vaeng**
Two exercises cover topics / concept, that are not (yet) in the syllabus, so I gave them a bit more implementation hints in the `instructions.append.md`

### CODE REVIEW

**Review by BethanyG — APPROVED** [review]
Going to approve with the warning that I think the `# Instructions append` is needed on both files per [the spec](https://exercism.org/docs/building/tracks/practice-exercises).  Not sure what happens if its missing, but the header may be needed for display, even if it is omitted when processed.

File: exercises/practice/allergies/.docs/instructions.append.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+This exercise uses the [`unordered_set`][set].
+You probably need the `emplace` function to add elements to the set.
```
[the remark this answers is not in the document]
Comment (vaeng) [reply exercises/practice/allergies/.docs/instructions.append.md:2]: Good addition!

File: exercises/practice/allergies/.docs/instructions.append.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+This exercise uses the [`unordered_set`][set].
```
[the remark this answers is not in the document]
Comment (vaeng) [reply exercises/practice/allergies/.docs/instructions.append.md:1]: Good to know!

**Comment by vaeng** [comment]
Let's merge and see if your suspicion is correct.

**Pull request merged**